pymeasure/display/inputs.py

```diff
@@ -210,8 +209,12 @@ def setValue(self, value):
             raise ValueError("Invalid choice for parameter. "
                              "Must be one of %s" % str(self._parameter.choices)) from e
 
+    def _addSuffix(self, value):
+        """ Private API to add suffix to each choice entry of the list """
+        self._stringChoices = tuple(choice + " " + str(value) for choice in self._stringChoices)
+
     def setSuffix(self, value):
-        self._stringChoices = tuple(choice + str(value) for choice in self._stringChoices)
+        pass
```

Comment:
I think the call of the setSuffix method is needed for the input widgets of class QSpinBox or QDoubleSpinBox (both they a setSuffix method) and it is needed to set the unit.
Since the other widget did not have this setSuffix method, the original coder implemented them as dummy calls, I guess.

Reply:
Yeah, if the intent was not to actively disable/no-op the setsuffix methods of the Qt superclass, I think we should be able to remove them now that all our implementations only have `pass`. @feinsteinben I think you touched that code last, any thoughts/insight on this?

Reply:
There does seem to be a case where the `setSuffix` method has to be implemented: whenever the `VectorParameter` or the `PhysicalParameter` (which, by the way, is not imported in `experiment.__init__.py`, unlike the other Parameters) are used, `units` is a valid attribute and therefore `setSuffix` will be called in `Input.set_parameter`.
And, as @msmttchr mentions, the call to `setSuffix` in `Input.set_parameter` is indeed required for correctly setting the unit of the parameter.

Reply:
OK, but now we have 3 empty `setSuffix` implementations, so how should we best correct that? It seems the logic around different combinations of Parameter subclasses and widget types, and which ones need suffixes when, is a bit unclear/gnarly - can we simplify that? I'll defer this decision to you (or Colin), as I am not really familiar with this part of the codebase.

Reply:
The best way that I can think of to make it more clear/convenient is to make sure there is a dedicated input for each type of in inputs (right now, only for the VectorParameter and the PhysicalParameter, which is derived from the VectorParameter there are no dedicated inputs).
As this is something that I think would fall a bit outside the scope of this PR, I think that for now there are a couple of options for solving the `pass` implementation: leave it be for now, add an additional check (`hasattr(self, 'setSuffix')`) to the if statement around the call to `setSuffix` in `set_parameter`, or move the call to `setSuffix` to a re-implementation of the `set_parameter` method in the specific sub-classes of `Input` that actually needs this call.
Although this last option seems to me rather nice (as you don't refer to a method that the class might not have), it results in a bit of duplicate code (though it would be only a single line in the `ScientificInput`, `FloatInput`, and `IntegerInput`).

Reply:
My two cents: I would leave it as it is for now, since some more restructuring could happen in the future and solve the problem.

Reply:
ok then, I created #365 to keep track of this, further inputs welcome over there.